### PR TITLE
Feature/app setup

### PR DESCRIPTION
**Pull request by mbenfowler**
## What this PR does / Testing notes
- Add main view `/` that shows birds in user's area
  - currently just a mocked set of bird codes, but a live fetch for those individual bird deets
- Each bird in this view can be selected with its respective checkbox
- When click on the bird button in the nav bar, you navigate to the `/saved` route
  - This displays just the birds that have been checked in the main view
  - A user may remove a bird from the watch list here as well
- Any invalid url will fallback to the `EmptyState` fallback (no content yet there yet)

### Notes
- I couldn't figure a way to `await` the results of `getBirdsData()` so I had to put an ugly `setTimeout` for the data to actually be in state... is there a better way to do this??
- Only styled for mobile so far

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
-
-    </div>
+    <>
+      <Nav />
+      <main className="App">
+        <Routes>
+          <Route path='/' element={<Home />}/>
+          <Route path='/saved' element={<Saved />}/>
+          <Route path='/settings' element={<Settings />}/>
+          <Route path="*" element={<EmptyState />} />
```
Comment (AlexaVanSchaardenburg) [inline src/components/App/App.js:20]: Nice path handling here!

File: src/apiCalls.js [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+    return await res.json()
+}
+
+const getBirdsData = (keys) => {
+    let birdsData = [];
+    keys.forEach(async key => {
+        let thisBirdData;
+        await getBirdData(key).then(result => thisBirdData = {...result[0], isChecked: false})
+        birdsData.push(thisBirdData)
+    });
```
Comment (jalbe0076) [inline src/apiCalls.js:16]: I read that async functions don't work with forEach loop, try using a for of loop instead, it worked on my end and you can then remove the setTimeout function.

File: src/components/App/App.js [hunk 3]
```diff
@@ -1,12 +1,55 @@
...
+      (async() => {
+          try {
+              // const birdKeys = await getBirdKeysByLocation(dummyLoc)
+              // const birdsData = await getBirdsData(birdKeys)
+              const birdsData = getBirdsData(mockBirdKeys)
+              setBirds(birdsData)
+              setTimeout(() => {
+                  setIsLoaded(true)
+              }, 1000)
+          } catch (error) {
```
Comment (jalbe0076) [inline src/components/App/App.js:29]: This setTimeout function can be changed if you update your API call loop

**Review by jalbe0076 — CHANGES_REQUESTED** [review]
Found a solution to your issue. async functions don't work with forEach loops, that's why you're not getting the data. It's because it does not pause the forEach each time it sees await.

File: src/apiCalls.js [hunk 4]
```diff
@@ -1,8 +1,26 @@
...
+    const res = await fetch(`https://api.ebird.org/v2/product/spplist/${region}`, {
+        headers: {
+            "X-eBirdApiToken": "f5n5l2qml9if"
+        }
+    })
+    return await res.json()
+}
+
+const getBirdsData = (keys) => {
+    let birdsData = [];
```
[the remark this answers is not in the document]
Comment (mbenfowler) [reply src/apiCalls.js:12]: yep it was totally the `forEach` that was causing this issue! Didn't get a chance to try the reduce, but a `for` loop is allowing me to async it now

**Pull request merged**